File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        long n=abs(x),d=0,s=0;
        while(n>0)
        {
            d=n%10;
            s=s*10+d;
            n=n/10;
        }
        cout<<s;
        if(s>(pow(2,31)-1))
        return 0;
        else
        {
            if(x>0)
              return s;
              else
              return -s;
        }
    }
};

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int res=0,d=0;
        int m= (x>=0)? x:x*-1; 
        while(m>0){
            d=m%10;
            if(res>(Integer.MAX_VALUE-d)/10)
                return 0;
            res=res*10+d;
            m=m/10;
        }
        return (x>0)?res: res*-1;
    }
}